keybase/client
# find convos - no send to pending

This replaces the old 'new conversation flow' with a simpler flow

### Old 
* Special pending conversation row
* chat-root level props for pending users, pending mode, if pending was selected
* Sending to pending would bookkeep creation and then sending, had a lot of code to deal with error cases if creation failed (retry etc)
* Search sagas had a lot of side effects and lack of side effects. Doing a search or cleaning it up required dispatching a series of actions in the correct order and making sure to never miss any

### New
* Special pending conversationidkey
* If this resolves to an existing real conversation we stash that conversationid key into the meta of the pending key
* Pending selected is just selectedConversation === Constants.pendingConversationIDKey
* Can't send into a pending, must always first create the conversation (really simplifies our side)
* Search sagas are more side effect drive. To exist a search just set pendingMode to 'none' etc
* Can exit search by clicking (X) in inbox item also

To test this:
Create conversations
Create and send to them
Search and exit
Reply privately 
etc

Make sure all the creation flows work correctly.
Make sure all the jumping into a conversation flows work correctly.
